### ISSUE TITLE
Sorting doesn't seem to work as expected?

### ISSUE THREAD

**Issue by ultrablue** [dbeniamine/todo.txt-vim#5]
Hello:

Sorting isn't working for me, at least as far as I understand the docs.

<LocalLeader>sp doesn't appear to do anything.

I'm trying to sort by project and then priority. Other sorts do seem to work.

I don't get an error, but the state of the buffer doesn't change either.

Thanks!
Greg

**Comment by dbeniamine** [comment]
Hello,

The mapping is `<LocalLeader>sp` which means `\sp` except if you have redefined your `<LocalLeader>` (cf  `:help <localleader>`).

`<LocalLeader>sp` is working fine for me, can you try to do:

    <LocalLeader>sc
    <LocalLeader>sp

If nothing happens there is indeed an issue, in this case can you tell me your
vim version, Operating system and give a todo.txt on which the bug occurs ?

Thanks,

David

**Comment by ultrablue** [comment]
Hi, David:

<LocalLeader> isn't set so it's \. That's working.

    :version
    VIM - Vi IMproved 7.4 (2013 Aug 10, compiled Dec  2 2015 10:34:56)
    MacOS X (unix) version
    Included patches: 1-944
    Compiled by Homebrew
    Huge version without GUI.  Features included (+) or not (-):

There's more; let me know if you need a more complete list.

Here's an example of what I'm seeing:

    (A) +aa_project Test @store
    (A) +ab_project A task.
    (A) +ab_project This is a task. @phone
    (B) +aa_project A task.
    (B) +aa_project Another task. @store
    (B) +ab_project Task. @phone

`<LocalLeader>s+`

    (B) +aa_project A task.
    (B) +aa_project Another task. @store
    (A) +aa_project Test @store
    (A) +ab_project A task.
    (B) +ab_project Task. @phone
    (A) +ab_project This is a task. @phone

`<LocalLeader>s`

    (A) +aa_project Test @store
    (A) +ab_project A task.
    (A) +ab_project This is a task. @phone
    (B) +aa_project A task.
    (B) +aa_project Another task. @store
    (B) +ab_project Task. @phone

`<LocalLeader>s@`

    (A) +ab_project A task.
    (B) +aa_project A task.
    (A) +ab_project This is a task. @phone
    (B) +ab_project Task. @phone
    (A) +aa_project Test @store
    (B) +aa_project Another task. @store

`<LocalLeader>s`

    (A) +aa_project Test @store
    (A) +ab_project A task.
    (A) +ab_project This is a task. @phone
    (B) +aa_project A task.
    (B) +aa_project Another task. @store
    (B) +ab_project Task. @phone

`<LocalLeader>sc`

    (A) +ab_project A task.
    (B) +aa_project A task.
    (A) +ab_project This is a task. @phone
    (B) +ab_project Task. @phone
    (A) +aa_project Test @store
    (B) +aa_project Another task. @store

`<LocalLeader>sp`

    (A) +ab_project A task.
    (B) +aa_project A task.
    (A) +ab_project This is a task. @phone
    (B) +ab_project Task. @phone
    (A) +aa_project Test @store
    (B) +aa_project Another task. @store






In other words, the last command, sp, didn't do anything.

Please let me know if there's anything else I can do.

Thanks!
Greg

**Comment by dbeniamine** [comment]
I can't reproduce the bug here ...

The main difference in our vim version is that I am on linux, but it does not relies on any unix command only pure vimscript ...

Do you have any todo.txt configuration that I should test ?

I've just committed a version of todo.txt-vim with debug info in the concerned function (and also a minor fix on groups detection that shouldn't impact this bug), can you update your todo.txt version, open a vim and run ":let g:Todo_txt_debug=1" then redo the last action aka run `<LocalLeader>sp`on the file:

    (A) +ab_project A task.
    (B) +aa_project A task.
    (A) +ab_project This is a task. @phone
    (B) +ab_project Task. @phone
    (A) +aa_project Test @store
    (B) +aa_project Another task. @store

You should see an output like this

    Linescount: 11
    Groups:
    ['ab_project', 'aa_project']
    execute sort i /.\{-}\ze+/
    execute 6,11sort i
    execute 1,11sort i

can you copy it here ?

Thanks,

David

**Comment by ultrablue** [comment]
Hi, David:

I'm not sure what you mean when you say 'Do you have any todo.txt configuration that I should test ?'

Linescount: 6
Groups:
['ab_project', 'aa_project']
execute sort i /.\{-}\ze+/
execute 4,6sort i
execute 1,3sort i

However, it looks like things are sorting better now.

It doesn't look like let g:Todo_txt_debug=0 is doing what I was hoping it would do. Is there a way to set the debug flag to false?

I'll keep you posted. I am seeing some odd behavior in my main todo file that could be attributed to formatting.

Thanks so much. I know how challenging finding edge-case bugs on opaque systems can be.

Greg

**Comment by dbeniamine** [comment]
Hi,

>  I'm not sure what you mean when you say 'Do you have any todo.txt configuration that I should test ?'

I meant that if you have added todo.txt configuration lines like:

    g:Todo_txt_second_level_sort_mode="i"

in your vimrc, I should try to reproduce the bug with these lines.

However I think that I understood the bug from your trace fixed it with commit a03d01e58ec1bc1a7c599210aa5ed5edd7303c5a. It seems that the linescount is wrong, I was indeed using a weird way to get it.

Can you pull the last version of todo.txt-vim and try to reproduce the bug (in debug mode) to see if it's ok (linescount should be 11 not 6) ? 

Concerning the debug flags you can either restart vim or do `:unlet g:Todo_txt_debug`

Tell me if I can close the issue.

Thanks for reporting it,
David
